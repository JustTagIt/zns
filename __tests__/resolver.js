const Simulator = require('../lib/simulater/Simulator')
const {Resolver} = require('../lib/simulator_contracts')

function makeBaseSim() {
  const sim = new Simulator()
  const address = sim.CreateAccount(10 ** 15) // 100 ZIL
  return {
    sim,
    resolver: new Resolver(
      {sim, fromAddr: address, gasLimit: 10000},
      {owner: address},
    ),
    address,
  }
}

describe('resolver users', () => {
  it('should be able to set values', () => {
    const {sim, resolver, address} = makeBaseSim()

    resolver.f.set({fromAddr: address, gasLimit: 10000}, 'test', '0x7357')

    expect(sim.GetSnapShot(true)).toMatchSnapshot()
  })

  it('should be able to unset values', () => {
    const {sim, resolver, address} = makeBaseSim()

    resolver.f.set({fromAddr: address, gasLimit: 10000}, 'test', '0x7357')
    resolver.f.unset({fromAddr: address, gasLimit: 10000}, 'test', '0x7357')

    expect(sim.GetSnapShot(true)).toMatchSnapshot()
  })

  it('should fail to set values if not the resolver owner', () => {
    const {sim, resolver, address} = makeBaseSim()

    resolver.f.set(
      {fromAddr: sim.CreateAccount(10 ** 15), gasLimit: 10000},
      'test',
      '0x7357',
    )

    expect(sim.GetSnapShot(true)).toMatchSnapshot()
  })

  it('should fail to unset values if not the resolver owner', () => {
    const {sim, resolver, address} = makeBaseSim()

    resolver.f.set({fromAddr: address, gasLimit: 10000}, 'test', '0x7357')
    resolver.f.unset(
      {fromAddr: sim.CreateAccount(10 ** 15), gasLimit: 10000},
      'test',
      '0x7357',
    )

    expect(sim.GetSnapShot(true)).toMatchSnapshot()
  })

  it('should fail to unset values if no keys exist', () => {
    const {sim, resolver, address} = makeBaseSim()

    resolver.f.unset({fromAddr: address, gasLimit: 10000}, 'test', '0x7357')

    expect(sim.GetSnapShot(true)).toMatchSnapshot()
  })
})
