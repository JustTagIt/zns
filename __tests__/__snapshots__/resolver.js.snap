// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resolver users should be able to set values 1`] = `
Object {
  "accounts": Object {
    "0x47e7a20867f04c007c2b76e96ec49bf8d9398bd8": Object {
      "address": "0x47e7a20867f04c007c2b76e96ec49bf8d9398bd8",
      "balance": 0n,
      "code": undefined,
      "init": Array [
        Object {
          "type": "ByStr20",
          "value": "0xa56193e39780d268b02bf5d9387179cc7ac9884b",
          "vname": "owner",
        },
        Object {
          "type": "BNum",
          "value": "0",
          "vname": "_creation_block",
        },
        Object {
          "type": "Uint32",
          "value": "0",
          "vname": "_scilla_version",
        },
        Object {
          "type": "ByStr20",
          "value": "0x47e7a20867f04c007c2b76e96ec49bf8d9398bd8",
          "vname": "_this_address",
        },
      ],
      "input": "/Users/laptop3/ZilliqaNameService/contracts/scilla/resolver.scilla",
      "state": Array [
        Object {
          "type": "Uint128",
          "value": "0",
          "vname": "_balance",
        },
        Object {
          "type": "Map (String) (ByStr)",
          "value": Array [
            Object {
              "key": "test",
              "val": "0x7357",
            },
          ],
          "vname": "records",
        },
      ],
    },
    "0xa56193e39780d268b02bf5d9387179cc7ac9884b": Object {
      "address": "0xa56193e39780d268b02bf5d9387179cc7ac9884b",
      "balance": 999999998018197n,
      "smartContracts": Array [
        "0x47e7a20867f04c007c2b76e96ec49bf8d9398bd8",
      ],
      "transactions": Array [
        "0xc51049598fbdffe2d3edd5aa1210aa9beff32faf27684afa115457a524f414c5",
        "0x2199c041141afe9e4de9d1900a28c3659538073b66d0eedb53525a3afb142627",
      ],
    },
  },
  "blockNumber": 0,
  "blocks": Array [],
  "currentBlock": Array [
    "0xc51049598fbdffe2d3edd5aa1210aa9beff32faf27684afa115457a524f414c5",
    "0x2199c041141afe9e4de9d1900a28c3659538073b66d0eedb53525a3afb142627",
  ],
  "events": Object {
    "0x111b38d3a1ba5fd8917dddfe1a993a4b87e498c04e7836a26c52f42241c16b4f": Object {
      "_eventname": "RecordSet",
      "address": "0x47e7a20867f04c007c2b76e96ec49bf8d9398bd8",
      "params": Array [
        Object {
          "type": "String",
          "value": "test",
          "vname": "key",
        },
        Object {
          "type": "ByStr",
          "value": "0x7357",
          "vname": "value",
        },
      ],
    },
  },
  "transactions": Object {
    "0x2199c041141afe9e4de9d1900a28c3659538073b66d0eedb53525a3afb142627": Object {
      "ID": "0x2199c041141afe9e4de9d1900a28c3659538073b66d0eedb53525a3afb142627",
      "amount": "0",
      "fromAddr": "0xa56193e39780d268b02bf5d9387179cc7ac9884b",
      "gasLimit": "10000",
      "gasPrice": "100",
      "meta": Object {
        "events": Array [
          "0x111b38d3a1ba5fd8917dddfe1a993a4b87e498c04e7836a26c52f42241c16b4f",
        ],
      },
      "nonce": "2",
      "receipt": Object {
        "cumulative_gas": "346",
        "epoch_num": "1",
        "success": true,
      },
      "toAddr": "0x47e7a20867f04c007c2b76e96ec49bf8d9398bd8",
      "version": "30134273",
    },
    "0xc51049598fbdffe2d3edd5aa1210aa9beff32faf27684afa115457a524f414c5": Object {
      "ID": "0xc51049598fbdffe2d3edd5aa1210aa9beff32faf27684afa115457a524f414c5",
      "amount": "0",
      "fromAddr": "0xa56193e39780d268b02bf5d9387179cc7ac9884b",
      "gasLimit": "10000",
      "gasPrice": "100",
      "meta": Object {
        "contractAddress": "0x47e7a20867f04c007c2b76e96ec49bf8d9398bd8",
      },
      "nonce": "1",
      "receipt": Object {
        "cumulative_gas": "991457",
        "epoch_num": "0",
        "success": true,
      },
      "toAddr": "0x0000000000000000000000000000000000000000",
      "version": "30134273",
    },
  },
}
`;

exports[`resolver users should be able to unset values 1`] = `
Object {
  "accounts": Object {
    "0x47e7a20867f04c007c2b76e96ec49bf8d9398bd8": Object {
      "address": "0x47e7a20867f04c007c2b76e96ec49bf8d9398bd8",
      "balance": 0n,
      "code": undefined,
      "init": Array [
        Object {
          "type": "ByStr20",
          "value": "0xa56193e39780d268b02bf5d9387179cc7ac9884b",
          "vname": "owner",
        },
        Object {
          "type": "BNum",
          "value": "0",
          "vname": "_creation_block",
        },
        Object {
          "type": "Uint32",
          "value": "0",
          "vname": "_scilla_version",
        },
        Object {
          "type": "ByStr20",
          "value": "0x47e7a20867f04c007c2b76e96ec49bf8d9398bd8",
          "vname": "_this_address",
        },
      ],
      "input": "/Users/laptop3/ZilliqaNameService/contracts/scilla/resolver.scilla",
      "state": Array [
        Object {
          "type": "Uint128",
          "value": "0",
          "vname": "_balance",
        },
        Object {
          "type": "Map (String) (ByStr)",
          "value": Array [],
          "vname": "records",
        },
      ],
    },
    "0xa56193e39780d268b02bf5d9387179cc7ac9884b": Object {
      "address": "0xa56193e39780d268b02bf5d9387179cc7ac9884b",
      "balance": 999999997027914n,
      "smartContracts": Array [
        "0x47e7a20867f04c007c2b76e96ec49bf8d9398bd8",
      ],
      "transactions": Array [
        "0xc51049598fbdffe2d3edd5aa1210aa9beff32faf27684afa115457a524f414c5",
        "0x2199c041141afe9e4de9d1900a28c3659538073b66d0eedb53525a3afb142627",
        "0xc6a0ee971806205e155c2229b34fb4c60e146da684005e8618cf5b1eb04c3082",
      ],
    },
  },
  "blockNumber": 0,
  "blocks": Array [],
  "currentBlock": Array [
    "0xc51049598fbdffe2d3edd5aa1210aa9beff32faf27684afa115457a524f414c5",
    "0x2199c041141afe9e4de9d1900a28c3659538073b66d0eedb53525a3afb142627",
    "0xc6a0ee971806205e155c2229b34fb4c60e146da684005e8618cf5b1eb04c3082",
  ],
  "events": Object {
    "0x111b38d3a1ba5fd8917dddfe1a993a4b87e498c04e7836a26c52f42241c16b4f": Object {
      "_eventname": "RecordSet",
      "address": "0x47e7a20867f04c007c2b76e96ec49bf8d9398bd8",
      "params": Array [
        Object {
          "type": "String",
          "value": "test",
          "vname": "key",
        },
        Object {
          "type": "ByStr",
          "value": "0x7357",
          "vname": "value",
        },
      ],
    },
    "0xe28330792789071f66770eb479dd457eaf7e2911f6d2ca94cbe02d6e9a445029": Object {
      "_eventname": "RecordUnset",
      "address": "0x47e7a20867f04c007c2b76e96ec49bf8d9398bd8",
      "params": Array [
        Object {
          "type": "String",
          "value": "test",
          "vname": "key",
        },
      ],
    },
  },
  "transactions": Object {
    "0x2199c041141afe9e4de9d1900a28c3659538073b66d0eedb53525a3afb142627": Object {
      "ID": "0x2199c041141afe9e4de9d1900a28c3659538073b66d0eedb53525a3afb142627",
      "amount": "0",
      "fromAddr": "0xa56193e39780d268b02bf5d9387179cc7ac9884b",
      "gasLimit": "10000",
      "gasPrice": "100",
      "meta": Object {
        "events": Array [
          "0x111b38d3a1ba5fd8917dddfe1a993a4b87e498c04e7836a26c52f42241c16b4f",
        ],
      },
      "nonce": "2",
      "receipt": Object {
        "cumulative_gas": "346",
        "epoch_num": "1",
        "success": true,
      },
      "toAddr": "0x47e7a20867f04c007c2b76e96ec49bf8d9398bd8",
      "version": "30134273",
    },
    "0xc51049598fbdffe2d3edd5aa1210aa9beff32faf27684afa115457a524f414c5": Object {
      "ID": "0xc51049598fbdffe2d3edd5aa1210aa9beff32faf27684afa115457a524f414c5",
      "amount": "0",
      "fromAddr": "0xa56193e39780d268b02bf5d9387179cc7ac9884b",
      "gasLimit": "10000",
      "gasPrice": "100",
      "meta": Object {
        "contractAddress": "0x47e7a20867f04c007c2b76e96ec49bf8d9398bd8",
      },
      "nonce": "1",
      "receipt": Object {
        "cumulative_gas": "991457",
        "epoch_num": "0",
        "success": true,
      },
      "toAddr": "0x0000000000000000000000000000000000000000",
      "version": "30134273",
    },
    "0xc6a0ee971806205e155c2229b34fb4c60e146da684005e8618cf5b1eb04c3082": Object {
      "ID": "0xc6a0ee971806205e155c2229b34fb4c60e146da684005e8618cf5b1eb04c3082",
      "amount": "0",
      "fromAddr": "0xa56193e39780d268b02bf5d9387179cc7ac9884b",
      "gasLimit": "10000",
      "gasPrice": "100",
      "meta": Object {
        "events": Array [
          "0xe28330792789071f66770eb479dd457eaf7e2911f6d2ca94cbe02d6e9a445029",
        ],
      },
      "nonce": "3",
      "receipt": Object {
        "cumulative_gas": "283",
        "epoch_num": "2",
        "success": true,
      },
      "toAddr": "0x47e7a20867f04c007c2b76e96ec49bf8d9398bd8",
      "version": "30134273",
    },
  },
}
`;

exports[`resolver users should fail to set values if not the resolver owner 1`] = `
Object {
  "accounts": Object {
    "0x111b38d3a1ba5fd8917dddfe1a993a4b87e498c0": Object {
      "address": "0x111b38d3a1ba5fd8917dddfe1a993a4b87e498c0",
      "balance": 999999999009649n,
      "smartContracts": Array [],
      "transactions": Array [
        "0xe28330792789071f66770eb479dd457eaf7e2911f6d2ca94cbe02d6e9a445029",
      ],
    },
    "0x47e7a20867f04c007c2b76e96ec49bf8d9398bd8": Object {
      "address": "0x47e7a20867f04c007c2b76e96ec49bf8d9398bd8",
      "balance": 0n,
      "code": undefined,
      "init": Array [
        Object {
          "type": "ByStr20",
          "value": "0xa56193e39780d268b02bf5d9387179cc7ac9884b",
          "vname": "owner",
        },
        Object {
          "type": "BNum",
          "value": "0",
          "vname": "_creation_block",
        },
        Object {
          "type": "Uint32",
          "value": "0",
          "vname": "_scilla_version",
        },
        Object {
          "type": "ByStr20",
          "value": "0x47e7a20867f04c007c2b76e96ec49bf8d9398bd8",
          "vname": "_this_address",
        },
      ],
      "input": "/Users/laptop3/ZilliqaNameService/contracts/scilla/resolver.scilla",
      "state": Array [
        Object {
          "type": "Uint128",
          "value": "0",
          "vname": "_balance",
        },
        Object {
          "type": "Map (String) (ByStr)",
          "value": Array [],
          "vname": "records",
        },
      ],
    },
    "0xa56193e39780d268b02bf5d9387179cc7ac9884b": Object {
      "address": "0xa56193e39780d268b02bf5d9387179cc7ac9884b",
      "balance": 999999999008543n,
      "smartContracts": Array [
        "0x47e7a20867f04c007c2b76e96ec49bf8d9398bd8",
      ],
      "transactions": Array [
        "0xc51049598fbdffe2d3edd5aa1210aa9beff32faf27684afa115457a524f414c5",
      ],
    },
  },
  "blockNumber": 0,
  "blocks": Array [],
  "currentBlock": Array [
    "0xc51049598fbdffe2d3edd5aa1210aa9beff32faf27684afa115457a524f414c5",
    "0xe28330792789071f66770eb479dd457eaf7e2911f6d2ca94cbe02d6e9a445029",
  ],
  "events": Object {
    "0x2199c041141afe9e4de9d1900a28c3659538073b66d0eedb53525a3afb142627": Object {
      "_eventname": "Error",
      "address": "0x47e7a20867f04c007c2b76e96ec49bf8d9398bd8",
      "params": Array [
        Object {
          "type": "Uint64",
          "value": "1",
          "vname": "code",
        },
        Object {
          "type": "String",
          "value": "Sender not owner",
          "vname": "message",
        },
      ],
    },
  },
  "transactions": Object {
    "0xc51049598fbdffe2d3edd5aa1210aa9beff32faf27684afa115457a524f414c5": Object {
      "ID": "0xc51049598fbdffe2d3edd5aa1210aa9beff32faf27684afa115457a524f414c5",
      "amount": "0",
      "fromAddr": "0xa56193e39780d268b02bf5d9387179cc7ac9884b",
      "gasLimit": "10000",
      "gasPrice": "100",
      "meta": Object {
        "contractAddress": "0x47e7a20867f04c007c2b76e96ec49bf8d9398bd8",
      },
      "nonce": "1",
      "receipt": Object {
        "cumulative_gas": "991457",
        "epoch_num": "0",
        "success": true,
      },
      "toAddr": "0x0000000000000000000000000000000000000000",
      "version": "30134273",
    },
    "0xe28330792789071f66770eb479dd457eaf7e2911f6d2ca94cbe02d6e9a445029": Object {
      "ID": "0xe28330792789071f66770eb479dd457eaf7e2911f6d2ca94cbe02d6e9a445029",
      "amount": "0",
      "fromAddr": "0x111b38d3a1ba5fd8917dddfe1a993a4b87e498c0",
      "gasLimit": "10000",
      "gasPrice": "100",
      "meta": Object {
        "events": Array [
          "0x2199c041141afe9e4de9d1900a28c3659538073b66d0eedb53525a3afb142627",
        ],
      },
      "nonce": "1",
      "receipt": Object {
        "cumulative_gas": "351",
        "epoch_num": "1",
        "success": true,
      },
      "toAddr": "0x47e7a20867f04c007c2b76e96ec49bf8d9398bd8",
      "version": "30134273",
    },
  },
}
`;

exports[`resolver users should fail to unset values if no keys exist 1`] = `
Object {
  "accounts": Object {
    "0x47e7a20867f04c007c2b76e96ec49bf8d9398bd8": Object {
      "address": "0x47e7a20867f04c007c2b76e96ec49bf8d9398bd8",
      "balance": 0n,
      "code": undefined,
      "init": Array [
        Object {
          "type": "ByStr20",
          "value": "0xa56193e39780d268b02bf5d9387179cc7ac9884b",
          "vname": "owner",
        },
        Object {
          "type": "BNum",
          "value": "0",
          "vname": "_creation_block",
        },
        Object {
          "type": "Uint32",
          "value": "0",
          "vname": "_scilla_version",
        },
        Object {
          "type": "ByStr20",
          "value": "0x47e7a20867f04c007c2b76e96ec49bf8d9398bd8",
          "vname": "_this_address",
        },
      ],
      "input": "/Users/laptop3/ZilliqaNameService/contracts/scilla/resolver.scilla",
      "state": Array [
        Object {
          "type": "Uint128",
          "value": "0",
          "vname": "_balance",
        },
        Object {
          "type": "Map (String) (ByStr)",
          "value": Array [],
          "vname": "records",
        },
      ],
    },
    "0xa56193e39780d268b02bf5d9387179cc7ac9884b": Object {
      "address": "0xa56193e39780d268b02bf5d9387179cc7ac9884b",
      "balance": 999999998018212n,
      "smartContracts": Array [
        "0x47e7a20867f04c007c2b76e96ec49bf8d9398bd8",
      ],
      "transactions": Array [
        "0xc51049598fbdffe2d3edd5aa1210aa9beff32faf27684afa115457a524f414c5",
        "0x2199c041141afe9e4de9d1900a28c3659538073b66d0eedb53525a3afb142627",
      ],
    },
  },
  "blockNumber": 0,
  "blocks": Array [],
  "currentBlock": Array [
    "0xc51049598fbdffe2d3edd5aa1210aa9beff32faf27684afa115457a524f414c5",
    "0x2199c041141afe9e4de9d1900a28c3659538073b66d0eedb53525a3afb142627",
  ],
  "events": Object {
    "0x111b38d3a1ba5fd8917dddfe1a993a4b87e498c04e7836a26c52f42241c16b4f": Object {
      "_eventname": "Error",
      "address": "0x47e7a20867f04c007c2b76e96ec49bf8d9398bd8",
      "params": Array [
        Object {
          "type": "Uint64",
          "value": "1",
          "vname": "code",
        },
        Object {
          "type": "String",
          "value": "Sender not owner or key does not exist",
          "vname": "message",
        },
      ],
    },
  },
  "transactions": Object {
    "0x2199c041141afe9e4de9d1900a28c3659538073b66d0eedb53525a3afb142627": Object {
      "ID": "0x2199c041141afe9e4de9d1900a28c3659538073b66d0eedb53525a3afb142627",
      "amount": "0",
      "fromAddr": "0xa56193e39780d268b02bf5d9387179cc7ac9884b",
      "gasLimit": "10000",
      "gasPrice": "100",
      "meta": Object {
        "events": Array [
          "0x111b38d3a1ba5fd8917dddfe1a993a4b87e498c04e7836a26c52f42241c16b4f",
        ],
      },
      "nonce": "2",
      "receipt": Object {
        "cumulative_gas": "331",
        "epoch_num": "1",
        "success": true,
      },
      "toAddr": "0x47e7a20867f04c007c2b76e96ec49bf8d9398bd8",
      "version": "30134273",
    },
    "0xc51049598fbdffe2d3edd5aa1210aa9beff32faf27684afa115457a524f414c5": Object {
      "ID": "0xc51049598fbdffe2d3edd5aa1210aa9beff32faf27684afa115457a524f414c5",
      "amount": "0",
      "fromAddr": "0xa56193e39780d268b02bf5d9387179cc7ac9884b",
      "gasLimit": "10000",
      "gasPrice": "100",
      "meta": Object {
        "contractAddress": "0x47e7a20867f04c007c2b76e96ec49bf8d9398bd8",
      },
      "nonce": "1",
      "receipt": Object {
        "cumulative_gas": "991457",
        "epoch_num": "0",
        "success": true,
      },
      "toAddr": "0x0000000000000000000000000000000000000000",
      "version": "30134273",
    },
  },
}
`;

exports[`resolver users should fail to unset values if not the resolver owner 1`] = `
Object {
  "accounts": Object {
    "0x47e7a20867f04c007c2b76e96ec49bf8d9398bd8": Object {
      "address": "0x47e7a20867f04c007c2b76e96ec49bf8d9398bd8",
      "balance": 0n,
      "code": undefined,
      "init": Array [
        Object {
          "type": "ByStr20",
          "value": "0xa56193e39780d268b02bf5d9387179cc7ac9884b",
          "vname": "owner",
        },
        Object {
          "type": "BNum",
          "value": "0",
          "vname": "_creation_block",
        },
        Object {
          "type": "Uint32",
          "value": "0",
          "vname": "_scilla_version",
        },
        Object {
          "type": "ByStr20",
          "value": "0x47e7a20867f04c007c2b76e96ec49bf8d9398bd8",
          "vname": "_this_address",
        },
      ],
      "input": "/Users/laptop3/ZilliqaNameService/contracts/scilla/resolver.scilla",
      "state": Array [
        Object {
          "type": "Uint128",
          "value": "0",
          "vname": "_balance",
        },
        Object {
          "type": "Map (String) (ByStr)",
          "value": Array [
            Object {
              "key": "test",
              "val": "0x7357",
            },
          ],
          "vname": "records",
        },
      ],
    },
    "0xa56193e39780d268b02bf5d9387179cc7ac9884b": Object {
      "address": "0xa56193e39780d268b02bf5d9387179cc7ac9884b",
      "balance": 999999998018197n,
      "smartContracts": Array [
        "0x47e7a20867f04c007c2b76e96ec49bf8d9398bd8",
      ],
      "transactions": Array [
        "0xc51049598fbdffe2d3edd5aa1210aa9beff32faf27684afa115457a524f414c5",
        "0x2199c041141afe9e4de9d1900a28c3659538073b66d0eedb53525a3afb142627",
      ],
    },
    "0xe28330792789071f66770eb479dd457eaf7e2911": Object {
      "address": "0xe28330792789071f66770eb479dd457eaf7e2911",
      "balance": 999999999009671n,
      "smartContracts": Array [],
      "transactions": Array [
        "0x1055e3f83f380244ae2ba92a2f4b3cc5d0dd814e27954321cee3bf824b5929a6",
      ],
    },
  },
  "blockNumber": 0,
  "blocks": Array [],
  "currentBlock": Array [
    "0xc51049598fbdffe2d3edd5aa1210aa9beff32faf27684afa115457a524f414c5",
    "0x2199c041141afe9e4de9d1900a28c3659538073b66d0eedb53525a3afb142627",
    "0x1055e3f83f380244ae2ba92a2f4b3cc5d0dd814e27954321cee3bf824b5929a6",
  ],
  "events": Object {
    "0x111b38d3a1ba5fd8917dddfe1a993a4b87e498c04e7836a26c52f42241c16b4f": Object {
      "_eventname": "RecordSet",
      "address": "0x47e7a20867f04c007c2b76e96ec49bf8d9398bd8",
      "params": Array [
        Object {
          "type": "String",
          "value": "test",
          "vname": "key",
        },
        Object {
          "type": "ByStr",
          "value": "0x7357",
          "vname": "value",
        },
      ],
    },
    "0xc6a0ee971806205e155c2229b34fb4c60e146da684005e8618cf5b1eb04c3082": Object {
      "_eventname": "Error",
      "address": "0x47e7a20867f04c007c2b76e96ec49bf8d9398bd8",
      "params": Array [
        Object {
          "type": "Uint64",
          "value": "1",
          "vname": "code",
        },
        Object {
          "type": "String",
          "value": "Sender not owner or key does not exist",
          "vname": "message",
        },
      ],
    },
  },
  "transactions": Object {
    "0x1055e3f83f380244ae2ba92a2f4b3cc5d0dd814e27954321cee3bf824b5929a6": Object {
      "ID": "0x1055e3f83f380244ae2ba92a2f4b3cc5d0dd814e27954321cee3bf824b5929a6",
      "amount": "0",
      "fromAddr": "0xe28330792789071f66770eb479dd457eaf7e2911",
      "gasLimit": "10000",
      "gasPrice": "100",
      "meta": Object {
        "events": Array [
          "0xc6a0ee971806205e155c2229b34fb4c60e146da684005e8618cf5b1eb04c3082",
        ],
      },
      "nonce": "1",
      "receipt": Object {
        "cumulative_gas": "329",
        "epoch_num": "2",
        "success": true,
      },
      "toAddr": "0x47e7a20867f04c007c2b76e96ec49bf8d9398bd8",
      "version": "30134273",
    },
    "0x2199c041141afe9e4de9d1900a28c3659538073b66d0eedb53525a3afb142627": Object {
      "ID": "0x2199c041141afe9e4de9d1900a28c3659538073b66d0eedb53525a3afb142627",
      "amount": "0",
      "fromAddr": "0xa56193e39780d268b02bf5d9387179cc7ac9884b",
      "gasLimit": "10000",
      "gasPrice": "100",
      "meta": Object {
        "events": Array [
          "0x111b38d3a1ba5fd8917dddfe1a993a4b87e498c04e7836a26c52f42241c16b4f",
        ],
      },
      "nonce": "2",
      "receipt": Object {
        "cumulative_gas": "346",
        "epoch_num": "1",
        "success": true,
      },
      "toAddr": "0x47e7a20867f04c007c2b76e96ec49bf8d9398bd8",
      "version": "30134273",
    },
    "0xc51049598fbdffe2d3edd5aa1210aa9beff32faf27684afa115457a524f414c5": Object {
      "ID": "0xc51049598fbdffe2d3edd5aa1210aa9beff32faf27684afa115457a524f414c5",
      "amount": "0",
      "fromAddr": "0xa56193e39780d268b02bf5d9387179cc7ac9884b",
      "gasLimit": "10000",
      "gasPrice": "100",
      "meta": Object {
        "contractAddress": "0x47e7a20867f04c007c2b76e96ec49bf8d9398bd8",
      },
      "nonce": "1",
      "receipt": Object {
        "cumulative_gas": "991457",
        "epoch_num": "0",
        "success": true,
      },
      "toAddr": "0x0000000000000000000000000000000000000000",
      "version": "30134273",
    },
  },
}
`;
